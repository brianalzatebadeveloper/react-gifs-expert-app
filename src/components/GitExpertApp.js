import React, { useState } from 'react';
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = () => {

  const [categories, setCategories] = useState(['the rock']);

  return (
    <>
     <h2>GifExpertApp</h2>
     <AddCategory setCategories={ setCategories } />
     <hr />
     <ol>
       { 
          categories.map( (category, i) => (
             <GifGrid 
                key={ category }
                category={ category } 
             />
          )) 
       }
     </ol>
    </>
  );

}

export default GifExpertApp;