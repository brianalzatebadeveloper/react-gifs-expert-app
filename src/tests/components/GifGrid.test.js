import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGits } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {

  const category = 'The Rock';
  let wrapper;

  beforeEach( () => {
    jest.clearAllMocks();

    useFetchGits.mockReturnValue({
      data: [],
      loading: true
    });

    wrapper = shallow( <GifGrid category={ category } /> );
  });

  test('Debe mostrar el componente correctamente', () => {
    
    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => { 

    const gifs = [{
      id: 'ABC123',
      title: 'Can You Smell The Rock GIF by WWE',
      url: 'https://media1.giphy.com/media/l0HUg6Ypas42ubkXu/giphy.gif?cid=0858df19exqupm9xvcfjarglgzwksmozb9542poktfd6af8i&rid=giphy.gif&ct=g'
    }]; 

    useFetchGits.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow( <GifGrid category={ category } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe( false );
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

  });

});