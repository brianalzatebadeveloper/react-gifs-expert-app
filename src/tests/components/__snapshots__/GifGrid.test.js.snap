// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    The Rock
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    The Rock
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="Can You Smell The Rock GIF by WWE"
      url="https://media1.giphy.com/media/l0HUg6Ypas42ubkXu/giphy.gif?cid=0858df19exqupm9xvcfjarglgzwksmozb9542poktfd6af8i&rid=giphy.gif&ct=g"
    />
  </div>
</Fragment>
`;
